test/Microsoft.AspNetCore.Rewrite.Tests/IISUrlRewrite/FileParserTests.cs

```diff
@@ -128,6 +128,37 @@ public void RuleParse_ParseMultipleRules()
             AssertUrlRewriteRuleEquality(expected, res);
         }
 
+        [Fact]
+        public void Should_parse_global_rules()
+        {
+            // arrange
+            var xml = @"<rewrite>
+                            <globalRules>
+                                <rule name=""httpsOnly"" patternSyntax=""ECMAScript"" stopProcessing=""true"">
+                                    <match url="".*"" />
+                                    <conditions logicalGrouping=""MatchAll"" trackAllCaptures=""false"">
+                                        <add input=""{HTTPS}"" pattern=""off"" />
+                                    </conditions>
+                                    <action type=""Redirect"" url=""https://{HTTP_HOST}{REQUEST_URI}"" />
+                                </rule>
+                            </globalRules>
+                            <rules>
+                                <rule name=""Rewrite to article.aspx"">
+                                    <match url = ""^article/([0-9]+)/([_0-9a-z-]+)"" />
+                                    <action type=""Rewrite"" url =""article.aspx?id={R:1}&amp;title={R:2}"" />
+                                </rule>
+                            </rules>
+                        </rewrite>";
+
+            // act
+            var rules = new UrlRewriteFileParser().Parse(new StringReader(xml));
+
+            // assert
+            Assert.Equal(2, rules.Count);
```

Comment:
not trying to be difficult but just want to point out that the point of the test is to assert that file parsing works correctly with regard to identifying global vs local rules. doing the test setup to assert rules are fully parsed correctly distracts from the tests goal and is also redundant with other tests that specifically assert other aspects of rule parsing. if you insist, i'll make the suggested change but it doesn't feel right to me.